lib/net/modules/dns.toit

```diff
@@ -209,3 +233,35 @@ class DnsQuery_:
         position_block.call position + 2
         return
     position_block.call position + 1
+
+/// Limit the size of the cache to avoid using too much memory.
+trim_cache_ -> none:
+  if CACHE_.size < MAX_CACHE_SIZE_: return
+
+  // Cache too big.  Start by removing entries where the TTL has
+  // expired.
+  now := Time.monotonic_us
+  CACHE_.filter --in_place: | key value |
+    value.end > now
+
+  // Set the limit a bit lower now - we want to remove at least one third of
+  // the entries when we trim the cache since it's an expensive operation.
+  if CACHE_.size < MAX_TRIMMED_CACHE_SIZE_: return
+
+  // Remove every second entry.
+  toggle := true
+  CACHE_.filter --in_place: | key value |
+    toggle = not toggle
+    toggle
```

Comment:
Would it make sense to remove either entries with low TTL, or to keep track of when items were last used?
In theory removing an entry and adding it again in the cache-lookup would make it such that the unused ones are in front.

Reply:
I don't want to remove the ones with low TTL, because it means some long-lived ones that are not used often can clog the cache.

It would probably be a good idea to remove the least recently used, but for now I want to keep code size down, and I think 99.9% of the time we won't hit this code at all - 30 entries is actually a lot for one process.